src/actions/socialAction.js

```diff
@@ -0,0 +1,40 @@
+import { toastr } from 'react-redux-toastr';
+import { globalAuthenticated, globalError } from './globalActions';
+import { MESSAGE } from '../utils/constants';
+
+/**
+ * Social login action handler
+ *
+ * @export
+ * @param {object} payload - The user payload object
+ * @returns {object}
+ */
+export default function socialLogin(payload) {
+  return dispatch => {
+    const requestOptions = {
+      method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          'Access-Control-Allow-Origin': '*',
+        },
+      body: JSON.stringify(payload)
+    };
+
+    return fetch(`${process.env.API_BASE_URL}/auth/social`, requestOptions)
+      .then(res => res.json())
```

Comment:
Good job @devdbrandy . Kindly make use of async/await here.

Reply:
Hi @Andraquin, thanks for the feedback. I have implemented the requested changes to maintain readability.